File: edexOsgi/build.edex/opt/esb/js/TableRequest.js
/**
 * Obtains an entry from a table.
 *
 * Usage:
 *   include("TableRequest.js");
 *   var query = new TableQuery("tableName");
 *   query.addParameter ("site", "Valley"); 
 *   query.addParameter ("type", "flood"); 
 *   var results = query.execute();
 */

/**
 * Class constructor.
 *
 * @param tableName (String) identifies the table 
 */
function TableRequest(database, table){
  this.database = database;
  this.tableName = table;
  this.queryResults = null;
  this.query = new TableQuery(this.database,this.tableName);
}

function _addParameter(name,value,operand)
{
  if(arguments.length==2)
  {
    this.query.addParameter(name,value);
  } 
  else
  {
    this.query.addParameter(name,value,operand);
  }
}

function _addList(name, value){
  this.query.addList(name, value);
}

function _setCount(count){
  this.query.setCount(count);
}

function _setSortValue(sortValue){
  this.query.setSortBy(sortValue);
}

function _setTableName(name){
  this.tableName = name;
}

function _setDatabase(name){
  this.database = name;
}

/**
 * Main action method. Queries the database and returns the XML
 * representation of a persistable data object. 
 *
 * @return (String) XML representation of a table on success
 *         (String) XML null response string on failure
 */
function _execute()
{
  this.queryResults = this.query.execute();
  if(this.queryResults == null || this.queryResults.size() == 0)
  {
    var response = new MakeResponseNull("Query returned 0 results.",
                                           this.query);
    return response.execute();
  }
  else
  {
    return this.makeXmlResponse();  
  }
}

function _makeXmlResponse()
{
  var response = new Array();
  for(i=0; i < this.queryResults.size(); i++)
  {	
    var makeResponse = new MakeResponseXml(this.queryResults.get(i));
    response[i] = makeResponse.execute();
  }
  return response;
}

TableRequest.prototype.execute = _execute;
TableRequest.prototype.makeXmlResponse = _makeXmlResponse;
TableRequest.prototype.addParameter = _addParameter;
TableRequest.prototype.addList = _addList;
TableRequest.prototype.setCount = _setCount;
TableRequest.prototype.setSortValue = _setSortValue;
TableRequest.prototype.setTableName = _setTableName;
TableRequest.prototype.setDatabase = _setDatabase;

File: edexOsgi/com.raytheon.edex.uengine/utility/edex_static/base/python/TableRequest.py
##
##

##
# This is a base file that is not intended to be overridden.
##

##
# uengine is deprecated and will be removed from the system soon. Migrate your
# apps to using the Data Access Framework (DAF).
##

#
# Request of table data 
#
#    
#     SOFTWARE HISTORY
#    
#    Date            Ticket#       Engineer       Description
#    ------------    ----------    -----------    --------------------------
#    04/17/08                      njensen       Initial Creation.
#    
#



from java.util import ArrayList
from com.raytheon.edex.uengine.tasks.query import TableQuery
from com.raytheon.edex.uengine.tasks.response import MakeResponseXml
from com.raytheon.edex.uengine.tasks.response import MakeResponseNull
from com.raytheon.uf.common.message.response import ResponseMessageGeneric


class TableRequest():
    def __init__(self, db, tbl):
        self.__database = db
        self.__table = tbl
        self.__queryResults = None
        self.__query = TableQuery(self.__database, self.__table)
    
    def addParameter(self, name, value, operand="="):
        self.__query.addParameter(name, value, operand)
    
    def addList(self, name, value):
        self.__query.addList(name, value)

    def setCount(self, count):
        self.__query.setCount(count)
    
    def setSortValue(self, sortValue):
        self.__query.setSortBy(sortValue)
    
    def setTableName(self, name):
        self.__table = name
    
    def setDatabase(self, name):
        self.__database = name
    
    def __makeXmlResponse(self):
        size = self.__queryResults.size()
        response = ArrayList()
        for i in range(size):
            response.add(ResponseMessageGeneric(self.__queryResults.get(i)))
        return response
    
    def execute(self):
        self.__queryResults = self.__query.execute()
        return self.__makeXmlResponse()
         